encoding/thrift/thriftrw-plugin-yarpc/server.go

```diff
@@ -130,6 +130,20 @@ func (h handler) <.Name>(ctx <$context>.Context, body <$wire>.Value) error {
 	return h.impl.<.Name>(ctx, <range .Arguments>args.<.Name>,<end>)
 }
 <else>
+
+func (h handler) <.Name>_applicationErrorName(err error) string {<if .Exceptions>
+    switch err.(type) {
+		<range .Exceptions ->
+		case <formatType .Type>:
+			if v, ok := err.(<formatType .Type>); ok {
```

Comment:
instead of adding an additional method and iterating through known exceptions, what if we just did an interface assertion in the handler logic for 
```
type ErrorNamer interface { 
 ErrorName() string
}

func (handler) Foo() ...
 ...
 if namer, ok := err.(ErrorNamer); ok {
  response.ApplicationErrorName = namer.ErrorName()
 }
}
```

I think that this will simplify the logic/generated code quite a bit

Reply:
This does in indeed simplify the generated code quite a bit. Thank you!